File: app/android/java/MoviesApp/app/src/main/java/com/example/moviesapp/view/components/listitem/ListItemComponentHolder.java
package com.example.moviesapp.view.components.listitem;

import androidx.annotation.NonNull;

import com.example.moviesapp.view.util.ComponentHolder;
import com.example.ui_components.BaseView;
import com.example.ui_components.databinding.ViewListItemBinding;
import com.example.ui_components.listitem.ListItemClickHandler;
import com.example.ui_components.listitem.model.ListItem;

/**
 * List item component holder.
 */
public class ListItemComponentHolder
        extends ComponentHolder<ViewListItemBinding, ListItem, ListItemClickHandler> {

    public ListItemComponentHolder(
            @NonNull BaseView<ViewListItemBinding, ListItem, ListItemClickHandler> view
    ) {
        super(view);
    }
}

File: app/android/kotlin/MoviesApp/app/src/main/java/com/example/moviesapp/view/components/listitem/ListItemComponentHolder.kt
package com.example.moviesapp.view.components.listitem

import com.example.moviesapp.view.util.ComponentHolder
import com.example.ui_components.BaseView
import com.example.ui_components.databinding.ViewListItemBinding
import com.example.ui_components.listitem.ListItemClickHandler
import com.example.ui_components.listitem.model.ListItem

/**
 * List item component holder.
 */
internal class ListItemComponentHolder(
        view: BaseView<ViewListItemBinding,
                ListItem,
                ListItemClickHandler>
) : ComponentHolder<ViewListItemBinding, ListItem, ListItemClickHandler>(
        view
)